### PR TITLE
Sensors - Add Sonar to Sparky target

### PR DESCRIPTION
**Pull request by JohnieBraaf**
To enable sonar for the sparky board, trigger (PWM6), echo (PWM7), I've made a few changes.

1) enable sonar (Makefile, target.h, sonar.c)
2) extend sonarHardware_s struct (added echo_gpio, trigger_gpio)
3) make switchable the GPIO pointer (to make it possible to use GPIOA pin 2)

All is working and tested. See https://onedrive.live.com/redir?resid=4416f39abab8e8a5%219294

### CODE REVIEW

**Comment by JohnieBraaf** [comment]
Ai, will need to make this include more flexible: #include "stm32f30x_gpio.h"
The build failed on not STM32F3 devices.

**Comment by JohnieBraaf** [comment]
This PR is ready to be merged.

File: src/main/drivers/sonar_hcsr04.c [hunk 1]
```diff
@@ -46,7 +46,7 @@ static void ECHO_EXTI_IRQHandler(void)
     static uint32_t timing_start;
     uint32_t timing_stop;
 
-    if (digitalIn(GPIOB, sonarHardware->echo_pin) != 0) {
+	if (digitalIn(sonarHardware->echo_gpio, sonarHardware->echo_pin) != 0) {
```
Comment (ledvinap) [inline src/main/drivers/sonar_hcsr04.c]: Please change indentation to 4 spaces ...

**Comment by JohnieBraaf** [comment]
Thnx, for the feedback, fixed, I will keep an eye on that ;)

**Comment by JohnieBraaf** [comment]
Woops accidently commited F4 library folder. Can I leave it?

**Comment by DaTwo** [comment]
Tested and working for me.

**Comment by JohnieBraaf** [comment]
@hydra  Would be nice if this would make it into 1.9

**Comment by hydra** [comment]
Merged with master, thanks @JohnieBraaf

**Pull request merged**